gammapy/maps/wcsnd.py

```diff
@@ -589,6 +585,63 @@ def get_spectrum(self, region=None, func=np.nansum, weights=None):
 
         return self.to_region_nd_map(region=region, func=func, weights=weights)
 
+    def binary_erosion(self, margin, from_edges=False):
+        """Binary erosion of boolean mask removing a given margin
+        Parameters
+        ----------
+        margin : tuple of `~astropy.units.Quantity`
+            Angular sizes of the margin in (lon, lat) in that specific order.
+            If only one value is passed, the same margin is applied in (lon, lat).
+        from_edge : bool
+            If True apply the margin only from edges (Default is False).
+        """
+        if self.data.dtype != bool:
+            raise (TypeError, "Binary erosion should be applied only on boolean mask")
+        if from_edges:
```

Comment:
This option I don't understand, I would expect `.binary_erosion()` to just act on `self.data` and not create any new mask. But we could e.g. offer something equivalent to `WcsGeom.region_mask()`, like `WcsGeom.boundary_mask(width=(lon, lat))` or similar...

Reply:
`from_edge` erode only the edges. The use case is the following :
 you have a dataset with mask_fit that mask some source already taking into account the psf, then you do a cutout in that case you want to apply the mask erosion only from the edges of the geom not based on the mask values.

Reply:
ok if you prefer to have it as a separate method